packages/service/lib/plugins/metrics.js

```diff
@@ -12,10 +12,12 @@ const metricsPlugin = fp(async function (app, opts = {}) {
   const register = new promClient.Registry()
 
   const defaultMetrics = opts.defaultMetrics ?? { enabled: true }
-  const prefix = opts.prefix ?? ''
 
-  if (opts.labels) {
+  if (opts.labels || opts.prefix) {
     const labels = opts.labels ?? {}
+    if (opts.prefix) {
+      labels.prefix = opts.prefix
```

Comment:
I would call it a `serviceId`

Reply:
Not sure, we already have this in the config as `prefix`. Which I know is generic, but in `service` we don't have the concept of `entrypoint`, etc...so I think it's OK to leave it as "prefix" and then set it in runtime (like we already do  here: https://github.com/platformatic/platformatic/blob/main/packages/runtime/lib/app.js#L305 )

Reply:
To clarify, in a runtime, this won't be the service id, but the entrypoint serviceId: https://github.com/platformatic/platformatic/blob/main/packages/runtime/lib/app.js#L304

Reply:
Why? Each service should attach it's own serviceId/prefix label and each runtime should attach an application name label. IMHO

Reply:
Well, in a runtime, the serviceId is already set as `prefix` ([here](https://github.com/platformatic/platformatic/blob/main/packages/runtime/lib/app.js#L305 )
But now it won't be used anymore to "prefix" metrics...so we could rename it in `serviceId`, indeed, but if we do so it would be a [breaking change](https://github.com/platformatic/platformatic/blob/main/packages/service/schema.json#L638C12-L640C15), and we could not backport these changes to 1.x

So what we should do IMO is to set the `serviceId` in `labels` config in runtime [here](https://github.com/platformatic/platformatic/blob/main/packages/runtime/lib/app.js#L305) 
If we do so we can safely leave the `prefix` in config and we will have the `serviceId` in prometheus metrics. 

WDYT @ivan-tymoshenko ?

Reply:
As I understand you need to drop the previx from the metric name for the ICC. So it would be a breacking change anyway. I'm ok with merging this changes into platformatic

Reply:
@ivan-tymoshenko PTAL. The prefix if set is now just set as label (we might remove it in a subsequent PR). In any case in runtime, the `serviceId` is now set and the automatically added to prom metrics. Added tests for this new part.

Reply:
I don't understand why do we need the prefix label. It's anyway a breacking change because it's not a metric name prefix, but a label, right? So it seems like we don't need the prefix config options anymore. If user want to set a custom prefix label, he can set it with custom prom labels.

Not sure I understand the idea of keeping it. Maybe I don't see some usecase.